src/jit/flowgraph.cpp

```diff
@@ -7440,7 +7440,7 @@ GenTreePtr Compiler::fgDoNormalizeOnStore(GenTreePtr tree)
 
             if (varDsc->lvNormalizeOnStore())
             {
-                noway_assert(op1->gtType <= TYP_INT);
+                noway_assert(genTypeSize(op1->gtType) <= genTypeSize(TYP_INT));
```

Comment:
This isn't needed, it's not about the type size but about checking for certain types - small ints and int. The assert is actually of questionable use, this code is executed only if `genActualType(op1->TypeGet())` is `TYP_INT` and that implies that `op1->TypeGet()` <= `TYP_INT`. Well, except if it is `TYP_UINT` but that should not appear on a node...

Reply:
So we can keep the original assert?

Reply:
Yes, it's not affected by the removal of `TYP_CHAR` anyway.